developer/guides/components/demo-component.mdx

```diff
@@ -0,0 +1,559 @@
+
+```tsx src/components/Accordions/Accordion.makeswift.tsx
+import { runtime } from "src/makeswift/runtime";
+
+import { Accordions } from "./";
+
+runtime.registerComponent(Accordions, {
+  type: "accordions",
+  label: "Accordions",
+  props: {},
+});
+```
+
+Learn more about [registering a component](/developer/guides/how-to/built-in-components#registering-the-component) in Makeswift.
+
+### Import the registered component
+
+Registered components need to be imported to show up in the Component Tray.
+
+We can copy the following line into our `src/makeswift/components.tsx` file, a file created during the installation step.
+
+```tsx src/makeswift/components.tsx
+import "src/components/Accordions/Accordions.makeswift";
+```
+
+Now, in Makeswift, we should see the `Accordions` component in the Component Toolbar, which we can then drag into the Canvas.
+
+Once we drag the component onto the Canvas, notice the `label` we defined in the `Accordions.makeswift.tsx` appears in the Properties Sidebar. Later, when we make the component editable in Makeswift, properties that can be edited will appear here.
+
+<Frame>
+  <video
+    autoPlay
+    muted
+    loop
+    playsInline
+    controls
+    title="Hello world component"
+    className="w-full aspect-video"
+    src="/images/developer/guides/components/demo-component/hello-world.mp4"
+  ></video>
+</Frame>
+
+### Build the Accordion component
+
+Let’s leverage the [Radix UI react accordion component](https://www.radix-ui.com/primitives/docs/components/accordion) to help us out. Stop your application and run the following command to install it:
+
+```
+npm install @radix-ui/react-accordion
+```
+
+In the `index.tsx` file, update your `Accordions` component to use the radix UI React accordion component. The example uses filler content for each accordion item. Each item has a title that triggers the expansion of its associated content.
+
+```tsx src/components/Accordions/index.tsx {1, 5-11, 12-19}
+import * as Accordion from "@radix-ui/react-accordion";
+
+export function Accordions() {
+  return (
+    <Accordion.Root type="single">
+      <Accordion.Item key="1" value={`item-${1}`}>
+        <Accordion.Trigger>Accordion 1</Accordion.Trigger>
+        <Accordion.Content>
+          Lorem ipsum dolor sit amet, consectetur adipiscing elit
+        </Accordion.Content>
+      </Accordion.Item>
+
+      <Accordion.Item key="2" value={`item-${2}`}>
+        <Accordion.Trigger>Accordion 2</Accordion.Trigger>
+        <Accordion.Content>
+          Lorem ipsum dolor sit amet, consectetur adipiscing elit
+        </Accordion.Content>
+      </Accordion.Item>
+    </Accordion.Root>
+  );
+}
+```
+
+Restart your development server. In the Visual Builder, when you click on the Accordion title, notice how the accordion doesn’t expand. That’s because we’re in Build mode by default. To toggle the accordion, we need to switch from Build mode to Interact mode in the Component Toolbar.
```

Comment:
Maybe
> Restart your development server. You'll see your changes reflected in the Makeswift Builder, but when you click the Accordion title, notice that it doesn’t expand. That’s because Makeswift starts in Build mode by default, which is designed for selecting components and editing their properties. To test behaviors like expanding the accordion, you need to be able to interact with the page as a user would. To do that, switch to Interact mode using the Component Toolbar.

?